### PR TITLE
[RW-10215][risk=low] Support localize Rmd and R files between disk and bucket.

### PR DESCRIPTION
**Pull request by yonghaoy**
This is similar to existing Jupyter file syncing, the only difference is the file pattern. (we want to sync R and Rmd now). 
Tested locally, both two way (GCS to disk and disk to GCS) works. 

---
**PR checklist**

- [ ] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/interactiveanalysis/InteractiveAnalysisService.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    localizeMap.put(aouConfigEditDir, aouConfigUri);
+    localizeMap.put(playgroundDir + "/" + AOU_CONFIG_FILENAME, aouConfigUri);
+
+    // Localize the requested notebooks, if any.
+    localizeMap.putAll(
+        fileNames.stream()
+            .collect(
+                Collectors.toMap(
+                    name -> localizeTargetDir + name, name -> gcsNotebooksDir + "/" + name)));
+    log.info(localizeMap.toString());
```
Comment (NehaBroad) [inline api/src/main/java/org/pmiops/workbench/interactiveanalysis/InteractiveAnalysisService.java:165]: Just curious We need to log this info?

File: api/src/main/resources/workbench-api.yaml [hunk 2]
```diff
@@ -6787,6 +6814,38 @@ definitions:
           which means it is a relative path in the Jupyter user-facing file system,
           e.g. "foo/bar.ipynb".'
 
+  AppLocalizeRequest:
+    type: object
+    required:
+    - notebookNames
```
Comment (NehaBroad) [inline api/src/main/resources/workbench-api.yaml:6820]: is this supposed to be fileNames?

File: api/src/main/resources/workbench-api.yaml [hunk 3]
```diff
@@ -6787,6 +6814,38 @@ definitions:
...
+        description: 'Names of notebooks to localize. This is just the basename (no
+          gs:// needed). This is interpreted as relative to the /notebooks directory
+          within the provided workspace''s Google Cloud Storage bucket.'
+        items:
+          type: string
+      playgroundMode:
+        type: boolean
+        default: false
+        description: Set to true if playgroundMode needed
+      appType:
```
Comment (NehaBroad) [inline api/src/main/resources/workbench-api.yaml:6834]: Do we need appType?

Comment (yonghaoy) [reply api/src/main/resources/workbench-api.yaml:6834]: AppType is not needed now, but would be needed once we support SAS. Adding appType here might be easier once we support more apps

File: api/src/main/java/org/pmiops/workbench/interactiveanalysis/InteractiveAnalysisService.java (hunk 1 above)
Comment (yonghaoy) [reply api/src/main/java/org/pmiops/workbench/interactiveanalysis/InteractiveAnalysisService.java:165]: It was added https://github.com/all-of-us/workbench/blob/0eeaa6f81a946a4bd3234f4a2b2a7b9b51bc7ea6/api/src/main/java/org/pmiops/workbench/api/RuntimeController.java#L370, I just copied it over. But I agree this is not useful. I will remove it. 
Thanks!

File: api/src/main/resources/workbench-api.yaml (hunk 2 above)
Comment (yonghaoy) [reply api/src/main/resources/workbench-api.yaml:6820]: ah, good catch! Thanks! Done

**Pull request merged**